### ISSUE TITLE
Feature request about build system update

### ISSUE THREAD

**Issue by Deophius** [Sudoku-dev/Sudoku#75]
Something about build system updates.
I think this should be better. Look at this:
```bash
[Deophius@localhost Sudoku]$ uptime
 14:05:50 up  1:55,  2 users,  load average: 0.70, 2.14, 2.16
[Deophius@localhost Sudoku]$ time cmake --build build
[  7%] Building CXX object History/CMakeFiles/history.dir/userconv.cpp.o
[ 14%] Linking CXX shared library libhistory.so
[ 14%] Built target history
[ 21%] Building CXX object Record/CMakeFiles/record.dir/alphabet.cpp.o
[ 28%] Building CXX object Record/CMakeFiles/record.dir/board.cpp.o
[ 35%] Building CXX object Record/CMakeFiles/record.dir/cell.cpp.o
[ 42%] Linking CXX shared library librecord.so
[ 42%] Built target record
[ 50%] Building CXX object Rule/CMakeFiles/rule.dir/field.cpp.o
[ 57%] Linking CXX shared library librule.so
[ 57%] Built target rule
[ 64%] Building CXX object Solve/CMakeFiles/solve.dir/puzzle.cpp.o
[ 71%] Building CXX object Solve/CMakeFiles/solve.dir/solve_result.cpp.o
[ 78%] Building CXX object Solve/CMakeFiles/solve.dir/solve.cpp.o
[ 85%] Linking CXX shared library libsolve.so
[ 85%] Built target solve
[ 92%] Building CXX object CMakeFiles/Sudoku.dir/test_main.cpp.o
[100%] Linking CXX executable Sudoku
[100%] Built target Sudoku

real	0m13.181s
user	0m12.262s
sys	0m0.883s
[Deophius@localhost Sudoku]$ uptime
 14:06:11 up  1:56,  2 users,  load average: 0.63, 2.04, 2.13
```
Simply too slow!

## Your best solution
Separate the building steps for each test file.

## Additional context
When there are other apps present, this would even be worse, like when the load average is above 1.

**Comment by Deophius** [comment]
Got it!
Look at this little example:

test1.cpp:
```cpp
#include <boost/test/unit_test.hpp>

BOOST_AUTO_TEST_CASE(case1)
{
    BOOST_CHECK(1 == 1);
}
```

test2.cpp:
```cpp
#include <boost/test/unit_test.hpp>

BOOST_AUTO_TEST_CASE(test2)
{
    BOOST_CHECK(2 == 2);
}
```

test3.cpp
```cpp
#include <boost/test/unit_test.hpp>

BOOST_AUTO_TEST_CASE(test3)
{
    BOOST_CHECK(3 == 3);
}
```

main.cpp
```cpp
#define BOOST_TEST_MODULE learn
#define BOOST_TEST_DYN_LINK
#include <boost/test/unit_test.hpp>

BOOST_AUTO_TEST_CASE(test4)
{
    BOOST_CHECK(4 == 4);
}
```

When compiled with this command:
```bash
g++ *.cpp -c -std=c++17 -Wall
g++ *.o -lboost_unit_test_framework -o learn
```

It correctly outputs an executable named `learn`.
Here is how it is run:
```bash
$ ./learn
Running 4 test cases...

*** No errors detected
```

So that's how we're going to update it.

**Comment by Deophius** [comment]
Resolved after merging in 425cb75 into file-mgr-2 !
